vip-api/src/main/java/fr/insalyon/creatis/vip/api/security/ApiSecurityConfig.java

```diff
@@ -36,141 +36,105 @@
 import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.security.web.AuthenticationEntryPoint;
-import org.springframework.security.web.authentication.session.NullAuthenticatedSessionStrategy;
-import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
+import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 import org.springframework.security.web.firewall.DefaultHttpFirewall;
-import org.springframework.stereotype.Service;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RegexRequestMatcher;
 
 import java.util.function.Supplier;
+import java.util.ArrayList;
 
 import static fr.insalyon.creatis.vip.api.CarminProperties.KEYCLOAK_ACTIVATED;
 
 /**
- * Keycloaksecurity configuration.
- *
- * It secures by api key all rest requests (except platform and authenticate)
- *
- * Modified by khalilkes to implement keycloak adapter
+ * VIP API configuration for API key and OIDC authentications.
  *
+ * Authenticates /rest requests with either:
+ * - a static per-user API key (ApikeyAuthenticationFilter)
+ * - or an OIDC Bearer token (ex-Keycloak). This part is currently work-in-progress,
+ *   with org.keycloak currently removed, and proper OIDC connector not implemented yet.
  */
-@ComponentScan(basePackageClasses = {KeycloakSecurityComponents.class})
-@KeycloakConfiguration
+@Configuration
 @EnableWebSecurity
-@Order(1)
-public class ApiSecurityConfig extends KeycloakWebSecurityConfigurerAdapter {
+public class ApiSecurityConfig {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     private final Environment env;
-    private final ApikeyAuthenticationProvider apikeyAuthenticationProvider;
     private final VipAuthenticationEntryPoint vipAuthenticationEntryPoint;
-    private final VipKeycloakAuthenticationProvider vipKeycloakAuthenticationProvider;
+    private final AuthenticationManager vipAuthenticationManager;
 
     @Autowired
     public ApiSecurityConfig(
             Environment env, ApikeyAuthenticationProvider apikeyAuthenticationProvider,
-            VipAuthenticationEntryPoint vipAuthenticationEntryPoint,
-            VipKeycloakAuthenticationProvider vipKeycloakAuthenticationProvider) {
+            VipAuthenticationEntryPoint vipAuthenticationEntryPoint) {
         this.env = env;
-        this.apikeyAuthenticationProvider = apikeyAuthenticationProvider;
         this.vipAuthenticationEntryPoint = vipAuthenticationEntryPoint;
-        this.vipKeycloakAuthenticationProvider = vipKeycloakAuthenticationProvider;
-    }
-
-    protected boolean isKeycloakActive() {
-        return env.getProperty(KEYCLOAK_ACTIVATED, Boolean.class, Boolean.FALSE);
-    }
-
-    @Override
-    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
```

Comment:
As far as I understand, this was needed for Bearer token authentication with the Keycloak connector, so I removed it as part of org.keycloak.* removal. I plan to add some equivalent back, if needed, for the new OIDC connector, but I do not have visibility on that yet.
If any equivalent is needed, it will for sure work differently in Spring6, as this old code was a method override of the parent `WebSecurityConfigurerAdapter` class that no longer exists, so keeping that as is would have been a no-op.